cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -280,83 +287,160 @@ static String removeReplicasFromConfig(String throttleConfig, Set<String> replic
-        LOG.debug("Existing config throttles all follower replicas. So, do not remove any follower replica throttle");
-      } else {
-        replicas.forEach(r -> LOG.debug("Removing follower throttles for topic {} and replica {}", topic, r));
-        String newThrottledReplicas = removeReplicasFromConfig(currentFollowerThrottledReplicas.value(), replicas);
-        if (newThrottledReplicas.isEmpty()) {
-          ops.add(new AlterConfigOp(new ConfigEntry(FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG, null), AlterConfigOp.OpType.DELETE));
+      if (currFollowerThrottle != null) {
+        if (currFollowerThrottle.source().equals(ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)) {
+          LOG.debug("Skipping removal for static follower throttle rate: {}", currFollowerThrottle);
         } else {
-          ops.add(new AlterConfigOp(new ConfigEntry(FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG, newThrottledReplicas), AlterConfigOp.OpType.SET));
+          LOG.debug("Removing follower throttle rate: {} on broker {}", currFollowerThrottle, brokerId);
+          ops.add(new AlterConfigOp(new ConfigEntry(FOLLOWER_REPLICATION_THROTTLED_RATE_CONFIG, null), AlterConfigOp.OpType.DELETE));
         }
       }
+      if (!ops.isEmpty()) {
+        bulkOps.put(cf, ops);
+      }
     }
-    if (!ops.isEmpty()) {
-      changeTopicConfigs(topic, ops);
+    applyIncrementalAlterConfigsForBrokers(bulkOps);
+  }
+
+  private void applyIncrementalAlterConfigsForBrokers(Map<ConfigResource, Collection<AlterConfigOp>> bulkOps)
+    throws ExecutionException, InterruptedException, TimeoutException {
+    if (bulkOps == null || bulkOps.isEmpty()) {
+      return;
     }
+    _adminClient.incrementalAlterConfigs(bulkOps).all().get(CLIENT_REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+    // Verify all broker configs in a single bulk describe
+    waitForConfigs(bulkOps);
   }
 
-  private void removeThrottledRateFromBroker(Integer brokerId)
-  throws ExecutionException, InterruptedException, TimeoutException {
-    Config brokerConfigs = getBrokerConfigs(brokerId);
-    ConfigEntry currLeaderThrottle = brokerConfigs.get(LEADER_REPLICATION_THROTTLED_RATE_CONFIG);
-    ConfigEntry currFollowerThrottle = brokerConfigs.get(FOLLOWER_REPLICATION_THROTTLED_RATE_CONFIG);
-    List<AlterConfigOp> ops = new ArrayList<>();
-    if (currLeaderThrottle != null) {
-      if (currLeaderThrottle.source().equals(ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)) {
-        LOG.debug("Skipping removal for static leader throttle rate: {}", currFollowerThrottle);
-      } else {
-        LOG.debug("Removing leader throttle rate: {} on broker {}", currLeaderThrottle, brokerId);
-        ops.add(new AlterConfigOp(new ConfigEntry(LEADER_REPLICATION_THROTTLED_RATE_CONFIG, null), AlterConfigOp.OpType.DELETE));
-      }
+  private void applyIncrementalAlterConfigsForTopics(Map<ConfigResource, Collection<AlterConfigOp>> bulkOps)
```

Comment:
This method is meant to replace `changeTopicConfigs`, right? If so, would it make sense to keep the method name a `changeTopicConfigs`?